src/model/ReplayModel.ts

```diff
@@ -0,0 +1,78 @@
+import fs from "fs";
+import path from "path";
+import { IModel } from "./IModel";
+import { IQueryResult } from "./IQueryResult";
+import { MetaInfo } from "../generator/MetaInfo";
+
+/**
+ * A model that reflects the prompts and completions observed during
+ * a previous execution.
+ */
+export class ReplayModel implements IModel {
+
+  private metaInfo: MetaInfo;
+  private promptToCompletionMap: Map<string, string> = new Map<string, string>();
```

Comment:
I don't think you need the type annotations since the `new` is fully annotated:

```suggestion
  private promptToCompletionMap = new Map<string, string>();
```